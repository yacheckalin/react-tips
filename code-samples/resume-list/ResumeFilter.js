import React from "react";
import PropTypes from "prop-types";

const ResumeFilter = ({ onFilter }) => {
  return (
    <div className="col s4 input-field">
      <span className="prefix">
        <i className="material-icons green-text">search</i>
      </span>
      <input
        type="text"
        className="validate"
        onChange={(e) => {
          onFilter(e.target.value);
        }}
      />
    </div>
  );
};

ResumeFilter.propTypes = {
  onFilter: PropTypes.func.isRequired,
};
export default ResumeFilter;
