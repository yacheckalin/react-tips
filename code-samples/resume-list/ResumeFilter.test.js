import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import ResumeFilter from "./ResumeFilter";

describe("<ResumeFilter />", () => {
  const props = {
    onFilter: jest.fn(),
  };
  test("renders and displays properly", () => {
    const wrapper = shallow(<ResumeFilter {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
