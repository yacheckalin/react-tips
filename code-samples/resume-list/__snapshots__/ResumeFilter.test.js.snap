// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumeFilter /> renders and displays properly 1`] = `
<Fragment>
  <div
    className="col s3 input-field"
  >
    <div>
      <label>
        <input
          checked={1}
          className="filled-in"
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <i
            className="material-icons red-text"
          >
            flag
          </i>
        </span>
      </label>
    </div>
    <div>
      <label>
        <input
          checked={1}
          className="filled-in"
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <i
            className="material-icons yellow-text"
          >
            flag
          </i>
        </span>
      </label>
    </div>
    <div>
      <label>
        <input
          checked={1}
          className="filled-in"
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <i
            className="material-icons green-text"
          >
            flag
          </i>
        </span>
      </label>
    </div>
  </div>
</Fragment>
`;
